ls1intum/Artemis
# Bugfix for LTI oauth and tasks/plantUmls in multiple programming exercises in exams

* Fix LTI oauth on the client side
* Fix issues with tasks/plantUmls in multiple programming exercises in exams
* Fix an issue when programming exercise repos are exported and the Jgit repository was not closed properly

### Checklist

### Motivation and Context
Bugfixes ;-)

### Description
1. Fix for LTI oauth: Make sure that the system notification component does not send a request to api/account
2. Fix for tasks: Refresh the markdown every time the user navigates between exercises and use unique ids (see plantUmls)
3. Fix for plantUmls: Use unique ids even when multiple problem statements are rendered in the DOM (on the same page but hidden in a different "tab")

Note: I also increased the plantUml cache which was not really used until now. This should prevent unnecessary REST calls to the server and improve the performance

**There is still an issue with results of one exercise affecting the task and uml status of other exercises. This is out of scope until further notice.**

### Steps for Testing
1. Try out an exam with multiple programming exercises but different tasks and plantUmls
2. Navigate between those
3. Tasks and PlantUmls should refresh correctly
Note: There is an ongoing exam on the test server 1 that you might want to join for testing
